### PR TITLE
feat(compiler)!: Custom box printing

### PR DESCRIPTION
**Pull request by spotandjake**
This pr adds custom box printing.
Now single argument tuples will be wrinted as `Box(1)` intead of `(1,)` given the only safe way to create a single arugment tuple is by using the box primitive.

I am wondering where I would add a regression test for this if needed I couldn't see anywhere, that we were testing `print` or `toString`.

Closes: #1620

### CODE REVIEW

File: stdlib/runtime/string.gr [hunk 1]
```diff
@@ -512,13 +513,12 @@ let rec heapValueToString = (ptr, extraIndents, toplevel) => {
...
         let rparen = ")"
+        let mut lparen = "("
         let mut strings = [rparen]
         if (tupleLength <= 1n) {
           // Special case: unary tuple, which is not valid Grain syntax; however, boxed values
           // are stored as a unary tuple, so we keep this in case one gets printed
-          let comma = ","
-          strings = [comma, ...strings]
-          void
+          lparen = "Box("
```
Comment (ospencer) [inline stdlib/runtime/string.gr:521]: It'd be a bit better if in this case you appended `"box"` to the list that's all going to get concatenated, so just `strings = ["box", ...strings]`. You might need to rework things a bit and push this `if` til after the loop. We'll also want to do lowercase `"box"` 🙂

**Review by ospencer — CHANGES_REQUESTED** [review]
Looking good! One small change, and then I think it'd be good to add one or two tests.

File: stdlib/runtime/string.gr (hunk 1 above)
Comment (spotandjake) [reply stdlib/runtime/string.gr]: Done this change.

**Comment by spotandjake** [comment]
> Looking good! One small change, and then I think it'd be good to add one or two tests.

I was wondering where I would put regression tests for these?

File: stdlib/runtime/string.gr [hunk 2]
```diff
@@ -28,6 +28,7 @@ include "runtime/unsafe/tags" as Tags
 include "runtime/numberUtils" as NumberUtils
 
 include "runtime/dataStructures" as DataStructures
+
```
Comment (ospencer) [inline stdlib/runtime/string.gr:31]:
```suggestion
```

**Comment by ospencer** [comment]
You can add tests to compiler/test/suites/strings.re!

**Review by ospencer — APPROVED** [review]
Sweet, thanks!

**Comment by ospencer** [comment]
I suppose this is technically breaking.

**Pull request merged**